tidb-cloud/data-service-manage-github-connection.md

```diff
@@ -0,0 +1,161 @@
+---
+title: Deploy Data App Automatically with GitHub
+summary: Learn how to deploy your Data App automatically with GitHub.
+---
+
+# Deploy Data App Automatically with GitHub
+
+TiDB Cloud provides a Configuration as Code (CaC) approach to represent your whole Data App configuration as code using the JSON syntax.
+
+By connecting your Data App to GitHub, TiDB Cloud can use the CaC approach and push your Data App configurations as [configuration files](/tidb-cloud/data-service-app-config-files.md) to your preferred GitHub repository and branch.
+
+If automatic deployment is enabled for your GitHub connection, you can also modify your Data App by updating its configuration files on GitHub. After you commit the configuration file changes to GitHub, the new configurations will be deployed in TiDB Cloud automatically.
+
+In this document, you will learn how to deploy your Data App automatically with GitHub and how to manage the GitHub connection.
+
+## Before you begin
+
+Before you connect a Data App to GitHub, make sure that you have the following:
+
+- A GitHub account.
+- A GitHub repository with your target branch.
+
+> **Note:**
+>
+> The GitHub repository is used to store [Data App configuration files](/tidb-cloud/data-service-app-config-files.md) after you connect a Data App to it. If the information (such as cluster ID and endpoint URL) in the configuration files is sensitive, make sure to use a private repository instead of a public one.
+
+## Step 1. Connect your Data App to GitHub
+
+You can connect your Data App to GitHub when you create the App. For more information, see [Create a Data App](/tidb-cloud/data-service-manage-data-app.md).
+
+If you did not enable the GitHub connection during the app creation, you can still enable it as follows:
+
+1. Navigate to the [**Data Service**](https://tidbcloud.com/console/data-service) page of your project.
+2. In the left pane, click the name of your target Data App to view its details.
+3. On the **Settings** tab, click **Connect** in the **Connect to GitHub** area. A dialog box for connection settings is displayed.
+4. In the dialog box, perform the following steps:
+
+    1. Click **Install on GitHub**, and then follow the on-screen instructions to install **TiDB Cloud Data Service** as an application on your target repository.
+    2. Click **Authorize** to authorize access to the application on GitHub.
+    3. Specify the target repository, branch, and directory where you want to store the configuration files of your Data App.
+
+        > **Note:**
+        >
+        > - The directory must start with a slash. For example, `/mydata`. If the directory you specified does not exist in the target repository and branch, it will be created automatically.
```

Comment:
```suggestion
        > - The directory must start with a slash (`/`). For example, `/mydata`. If the directory you specified does not exist in the target repository and branch, it will be created automatically.
```